### PR TITLE
12504 - Added code changes for FSS UI - Simplified search - Main sear…

### PR DESCRIPTION
**Pull request by pratibh14837**
PR contains Code changes and Unit test cases.

### CODE REVIEW

File: src/app/core/services/fss-search-filter.service.ts [hunk 1]
```diff
@@ -77,4 +77,20 @@ export class FssSearchFilterService {
     return operatorType
   }
 
+  getFilterExpressionForSimplifiedSearch(fssSearchFilter: string) {
+    let searchKeywords = fssSearchFilter.split(" ");
+     
+    var filterExpression = "";
```
Comment (crdandoukho) [inline src/app/core/services/fss-search-filter.service.ts:83]: Please prefer `let` and `const` variable declarations over `var` declarations

File: src/app/core/services/fss-search-filter.service.ts [hunk 2]
```diff
@@ -77,4 +77,20 @@ export class FssSearchFilterService {
...
+    for (var i in searchKeywords) {
+      if (searchKeywords[i] != "") {
```
Comment (crdandoukho) [inline src/app/core/services/fss-search-filter.service.ts:85]: Please prefer strict equality operators `===` & `!==`

File: src/app/core/services/fss-search-filter.service.ts [hunk 3]
```diff
@@ -77,4 +77,20 @@ export class FssSearchFilterService {
     return operatorType
   }
 
+  getFilterExpressionForSimplifiedSearch(fssSearchFilter: string) {
```
Comment (crdandoukho) [inline src/app/core/services/fss-search-filter.service.ts:80]: Please specify the return type
```suggestion
  getFilterExpressionForSimplifiedSearch(fssSearchFilter: string): string {
```

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Some changes needed, please see my comments.

I have also pushed a failing test to demonstrate the code is not removing all single quotes. Please fix the code to remove all single quotes.

File: src/app/core/services/fss-search-filter.service.ts (hunk 1 above)
Comment (pratibh14837) [reply src/app/core/services/fss-search-filter.service.ts]: Resolved

File: src/app/core/services/fss-search-filter.service.ts (hunk 2 above)
Comment (pratibh14837) [reply src/app/core/services/fss-search-filter.service.ts]: Resolved

File: src/app/core/services/fss-search-filter.service.ts (hunk 3 above)
Comment (pratibh14837) [reply src/app/core/services/fss-search-filter.service.ts]: Resolved

**Comment by pratibh14837** [comment]
we don't want to remove single quotes but to replace them with 2 single quotes and I have modified the code as same.

File: src/app/core/services/fss-search-filter.service.ts [hunk 4]
```diff
@@ -77,4 +77,20 @@ export class FssSearchFilterService {
     return operatorType
   }
 
+  getFilterExpressionForSimplifiedSearch(fssSearchFilter: string): string {
+    let searchKeywords = fssSearchFilter.split(" ");
+     
+    let filterExpression = "";
+    for (var i in searchKeywords) {
```
Comment (crdandoukho) [inline src/app/core/services/fss-search-filter.service.ts:84]: Please use `let` here
```suggestion
    for (let i in searchKeywords) {
```

File: src/app/core/services/fss-search-filter.service.ts [hunk 5]
```diff
@@ -77,4 +77,20 @@ export class FssSearchFilterService {
...
+      if (searchKeywords[i] !== "") {
+        var searchKeyword = searchKeywords[i].replace(/'/g, "''");
```
Comment (crdandoukho) [inline src/app/core/services/fss-search-filter.service.ts:86]: Please use `let` here
```suggestion
        let searchKeyword = searchKeywords[i].replace(/'/g, "''");
```

File: src/app/features/fss-search/fss-simplified-search/fss-simplified-search.component.ts [hunk 6]
```diff
@@ -25,6 +26,7 @@ export class FssSimplifiedSearchComponent implements OnInit {
 
   getSimplifiedSearchResult(){
     if (this.fieldValue.trim() !== "") {
+      var filterExpression=this.fssSearchFilterService.getFilterExpressionForSimplifiedSearch(this.fieldValue.trim());
```
Comment (crdandoukho) [inline src/app/features/fss-search/fss-simplified-search/fss-simplified-search.component.ts:29]: Please use `let` or `const` here
```suggestion
      let filterExpression=this.fssSearchFilterService.getFilterExpressionForSimplifiedSearch(this.fieldValue.trim());     
```

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Please use 'let' or 'const' instead of `var` to avoid unintended scope issues

File: src/app/core/services/fss-search-filter.service.ts (hunk 4 above)
Comment (pratibh14837) [reply src/app/core/services/fss-search-filter.service.ts]: resolved

File: src/app/core/services/fss-search-filter.service.ts (hunk 5 above)
Comment (pratibh14837) [reply src/app/core/services/fss-search-filter.service.ts]: resolved

File: src/app/features/fss-search/fss-simplified-search/fss-simplified-search.component.ts (hunk 6 above)
Comment (pratibh14837) [reply src/app/features/fss-search/fss-simplified-search/fss-simplified-search.component.ts]: resolved

**Review by crdandoukho — APPROVED** [review]
Looks good. Thanks

**Review by bhalcha13813 — APPROVED** [review]
LGTM

**Pull request merged**